Fixed TASK-823

sujata-m opened the pull request:
Fixed [Task-823](https://dev.azure.com/TDEI-UW/TDEI/_workitems/edit/823/)

sujata-m committed:
Fixed Task - 823

sujata-m committed:
Updated zip file

sujata-m committed:
Remove extra print statements

sureshgaussian commented on src/config.py:
```diff
@@ -11,6 +11,8 @@ class Settings(BaseSettings):
     incoming_topic_subscription: str = os.environ.get('CONFIDENCE_REQ_SUB', '')
     outgoing_topic_name: str = os.environ.get('CONFIDENCE_RES_TOPIC', '')
     storage_container_name: str = os.environ.get('CONTAINER_NAME', 'osw')
+    username: str = os.environ.get('USERNAME', 'dev@gaussiansolutions.com')
```
Please remove the default credentials from here. Never expose passwords in the code. If the USERNAME/PASSWORD are not provided, just exit the code saying those two are mandatory.

susrisha replied:
Please remove default credentials. we will be giving out our credentials here.

sureshgaussian commented on src/service/test.py:
what is this file? why is a test under src folder? If we release this file on public git, we should not have names like DifferentClass. Fix this.

sureshgaussian commented on src/service/osw_confidence_service.py:
```diff
@@ -44,39 +46,47 @@ def process(self, msg: QueueMessage):
                 jobId=jobId,
-                confidence_level='90.0',
+                confidence_level=score,
                 confidence_library_version='v1.0',
```
this should be the version of the osw_confidence_metrics library we are using; not a static number. As the code stands today I believe this should be `python_confidence_metric.__version__`
I see the python module for the confidence metrics library is named osw_confidence_metric (from requirements.txt), however the usage of the library is python_confidence_metric. This is not good. Module name and its usuage should be same. Change python_confidence_metric to osw_confidence_metric.

susrisha commented on src/service/osw_confidence_metric.py:
```diff
@@ -0,0 +1,62 @@
+        area_analyzer = AreaAnalyzer(osm_data_handler=osm_data_handler)
+        start_time = time.time()
+        score = area_analyzer.calculate_area_confidence_score(file_path=self.convex_file)
+        print("--- %s seconds ---" % (time.time() - start_time))
```
please use logger here.

susrisha commented on src/service/osw_confidence_metric.py:
```diff
@@ -0,0 +1,62 @@
+warnings.filterwarnings('ignore', category=DeprecationWarning)
+
+
+class OSWConfidenceMetric:
```
Write a couple of lines about what this class does in sequence

susrisha commented on src/service/osw_confidence_service.py:
```diff
@@ -44,39 +46,47 @@ def process(self, msg: QueueMessage):
         # The meta file is at `meta_file_local_path`
         # The osw file is at `osw_file_local_path`
         # insert code for calculation here.
+        metric = OSWConfidenceMetric(zip_file=osw_file_local_path)
```
If the confidence calculation only relies on osw file, comment out the download of meta file. Its an additional performance burden to download that file.

susrisha commented on src/service/test.py:
```diff
@@ -0,0 +1,21 @@
+from osw_confidence_metric  import OSWConfidenceMetric
+class DifferentClass:
```
Rename to `OSWConfidenceMetricTestClass`

susrisha commented on src/service/test.py:
```diff
@@ -0,0 +1,21 @@
+    def __init__(self):
+        pass
+
+    def my_function(self):
```
rename function to `test_osw_confidence_metric_calculation`

susrisha reviewed (changes requested):
Some changes needed in documenting and renaming stuff.

sujata-m committed:
Fixed review comments

sujata-m committed:
Updated confidence-metric package Version to 0.0.2

sujata-m committed:
Changed filename from `osw_confidence_metric.py` to `osw_confidence_metric_calculator.py` as it was conflicting with the confidence-metric package

sujata-m committed:
Change class name references from comments

sujata-m committed:
Updated code with following changes -

- Updated code to leverage __version__
- Fixed unit test cases
- Added few more unit test cases

sureshgaussian reviewed (approved):
approved after code review.
